Added tests and corrected names

alexanderrtaylor opened the pull request:
Added some more specific tests so that we know  better when it should/should not be turning on/off

Commit:
Added tests and corrected names

Added some more specific tests so that we know  better when it should/should not be turning on/off

bitwiseman commented on src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java:
```diff
@@ -13,10 +13,7 @@
-import org.junit.ClassRule;
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.*;
```
Revert this

Commit:
Update ApiRateLimitCheckerTest.java

Added a test and reverted some changes that were not needed

bitwiseman commented on src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java:
```diff
@@ -161,6 +160,12 @@ private void setupStubs(List<JsonRateLimit> scenarios) {
 
+    /**
+     * This test is designed to show when the throttle is not happening for "OnOver" throttle.
+     * So this is just after you have been handed a new quota and are asking for Github data
```
```suggestion
     * client has just been given a new quota.
```

bitwiseman commented on src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java:
```diff
@@ -161,6 +160,12 @@ private void setupStubs(List<JsonRateLimit> scenarios) {
     private Date soon = Date.from(LocalDateTime.now().plusMinutes(60).atZone(ZoneId.systemDefault()).toInstant());
 
+    /**
+     * This test is designed to show when the throttle is not happening for "OnOver" throttle.
```
```suggestion
     * Test that throttle does not happen for "ThrottleOnOver" when
```

alexanderrtaylor commented:
@julianvmodesto I am still working on this please dont merge it yet. I should have something by Monday to help out

Commit:
Test and code Corrections

Added some additional tests, adjusted existing tests and then added the correct sleeping time and set the expiration correctly

alexanderrtaylor commented:
@julianvmodesto I think this is OK  to merge but I will check with Liam (we may just push it)

julianvmodesto commented:
sgtm! Let me know if you'd like me to hit merge

alexanderrtaylor commented:
Sorry just a few more corrections because 2 tests were flakey and I want to correct

Commit:
Fixed flakey tests

Fixed some flakey tests and added some comments along with some additional cases

Commit:
Update ApiRateLimitCheckerTest.java

Added comments and fixed flakey test
